opencensus/stats/internal/stats_manager_benchmark.cc

```diff
@@ -110,11 +110,75 @@ BENCHMARK_TEMPLATE2(BM_Record, DistributionAggregation, CumulativeWindow)
 BENCHMARK_TEMPLATE2(BM_Record, DistributionAggregation, IntervalWindow)
     ->Range(1, 16);
 
+// Benchmarks batched recording against a set of measures with a small number of
+// views on each, matching RPC stats recording.
+void BM_RecordBatched(benchmark::State& state) {
+  const std::string tag_key_1 = "tag_key_1";
+  const std::string tag_key_2 = "tag_key_2";
+  const int num_measures = 6;
+  std::vector<MeasureDouble> measures;
+  std::vector<std::unique_ptr<View>> views;
+  for (int i = 0; i < num_measures; ++i) {
+    const std::string measure_name = MakeUniqueName();
+    measures.push_back(MeasureRegistry::RegisterDouble(measure_name, "", ""));
+
+    const ViewDescriptor count_descriptor =
+        ViewDescriptor()
+            .set_measure(measure_name)
+            .set_name(absl::StrCat("count_", measure_name))
+            .set_aggregation(Aggregation::Count())
+            .set_aggregation_window(AggregationWindow::Cumulative())
+            .add_column(tag_key_1)
+            .add_column(tag_key_2);
+    views.push_back(absl::make_unique<View>(count_descriptor));
+
+    const ViewDescriptor sum_descriptor =
+        ViewDescriptor()
+            .set_measure(measure_name)
+            .set_name(absl::StrCat("sum_", measure_name))
+            .set_aggregation(Aggregation::Sum())
+            .set_aggregation_window(AggregationWindow::Cumulative())
+            .add_column(tag_key_1)
+            .add_column(tag_key_2);
+    views.push_back(absl::make_unique<View>(sum_descriptor));
+
+    const ViewDescriptor distribution_descriptor =
+        ViewDescriptor()
+            .set_measure(measure_name)
+            .set_name(absl::StrCat("distribution_", measure_name))
+            .set_aggregation(Aggregation::Distribution(
+                BucketBoundaries::Exponential(10, 10, 2)))
+            .set_aggregation_window(AggregationWindow::Cumulative())
+            .add_column(tag_key_1)
+            .add_column(tag_key_2);
+    views.push_back(absl::make_unique<View>(distribution_descriptor));
+  }
+
+  std::vector<std::string> tag_values(10);
+  for (int i = 0; i < 10; ++i) {
+    tag_values[i] = absl::StrCat("value", i);
+  }
+  int iteration = 0;
```

Comment:
Optional: wrap everything above L162 in state.PauseTiming()/ResumeTiming()

Reply:
The benchmark library does not start timing until the first call to the state loop--as I understand PauseTiming()/ResumeTiming() are only needed if you are doing housekeeping inside the loop.